extensions/desktop-common/desktop-init

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash -e
+
+#################
+# Launcher init #
+#################
+
+# On Fedora $SNAP is under /var and there is some magic to map it to /snap.
+# # We need to handle that case and reset $SNAP
+SNAP="${SNAP/\/var\/lib\/snapd/}"
+
+SNAP_DESKTOP_COMPONENTS_NEED_UPDATE="true"
+
+# shellcheck source=/dev/null
+. "$SNAP_USER_DATA/.last_revision" 2>/dev/null || true
+if [ "$SNAP_DESKTOP_LAST_REVISION" = "$SNAP_REVISION" ]; then
+  SNAP_DESKTOP_COMPONENTS_NEED_UPDATE="false"
+fi
+
+# Set $REALHOME to the users real home directory
+REALHOME="$(getent passwd $UID | cut -d ':' -f 6)"
+
+# If the user has modified their user-dirs settings, force an update
+if [[ -f "$XDG_CONFIG_HOME/user-dirs.dirs.md5sum" && -f "$XDG_CONFIG_HOME/user-dirs.locale.md5sum" ]]; then
+  if [[ "$(md5sum < "$REALHOME/.config/user-dirs.dirs")" != "$(cat "$XDG_CONFIG_HOME/user-dirs.dirs.md5sum")" ||
+        "$(md5sum < "$REALHOME/.config/user-dirs.locale")" != "$(cat "$XDG_CONFIG_HOME/user-dirs.locale.md5sum")" ]]; then
+    SNAP_DESKTOP_COMPONENTS_NEED_UPDATE="true"
+  fi
+fi
+
+if [ "$SNAP_ARCH" == "amd64" ]; then
+  ARCH="x86_64-linux-gnu"
+elif [ "$SNAP_ARCH" == "armhf" ]; then
+  ARCH="arm-linux-gnueabihf"
+elif [ "$SNAP_ARCH" == "arm64" ]; then
+  ARCH="aarch64-linux-gnu"
+elif [ "$SNAP_ARCH" == "ppc64el" ]; then
+  ARCH="powerpc64le-linux-gnu"
+else
+  ARCH="$SNAP_ARCH-linux-gnu"
+fi
+
+SNAP_LAUNCHER_ARCH_TRIPLET="$ARCH"
+
+if [ -n "$SNAP_DESKTOP_RUNTIME" ]; then
+  WITH_SNAP_DESKTOP_RUNTIME="yes"
```

Comment:
My apologies in advance, as I lack quite a bit of context on this review, so I'll be making some silly questions.

In the case at hand, why do we have a variable stating that another variable is set?  Isn't the first variable being set in the first place enough?

Reply:
This is a throwback - I improved the code surrounding the use of `$SNAP_DESKTOP_RUNTIME` but didn't come back to swap out the tests and this variable. 364960e1 fixes it :-)